src/app/surveys/surveys.component.ts

```diff
@@ -102,6 +101,10 @@ export class SurveysComponent implements OnInit, AfterViewInit, OnDestroy {
     this.surveys.paginator = this.paginator;
   }
 
+  findSurveyInSteps(steps, survey) {
+    return steps.findIndex((step: any) => step.survey && step.survey._id === survey._id);
```

Comment:
I think rather than running this function many times you can revert to have it inside the `ngOnInit` and save the stepNum to the survey.